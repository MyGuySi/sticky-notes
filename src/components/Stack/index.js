// @flow

import * as React from 'react';
import cx from 'classnames';
import styles from './styles.module.css';

type Props = {
  direction?: 'row' | 'column',
  children: React.Node
}

function Stack (props: Props) {
  return (
    <div className={cx(styles.stack, {
      [styles.row]: props.direction === 'row'
    })}>
      {props.children}
    </div>
  )
}

export default Stack;