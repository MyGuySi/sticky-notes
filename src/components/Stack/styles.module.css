.stack > * + * {
  margin-top: 1rem;
}

.row {
  display: flex;
}

.row > * + * {
  margin-top: 0;
  margin-left: 1rem;
}