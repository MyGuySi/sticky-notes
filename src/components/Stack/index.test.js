// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import Stack from './index';

test('Renders correctly', () => {
  const component = renderer.create(
    <Stack>
      <p>One</p>
      <p>Two</p>
      <p>Three</p>
    </Stack>
  )
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})