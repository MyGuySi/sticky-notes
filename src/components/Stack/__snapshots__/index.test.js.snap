// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="stack"
>
  <p>
    One
  </p>
  <p>
    Two
  </p>
  <p>
    Three
  </p>
</div>
`;
